### DISCUSSION TITLE
Trained a new model from scratch (~90h of audio in Polish)

### DISCUSSION THREAD

**Discussion by piotrwitkowski** [SWivid/F5-TTS#961] [Show and tell]
https://www.youtube.com/watch?v=K6vY9Je4ufQ

It's funny to see how it converges.

**Comment by sheepmap** [comment]
Can you please share your experience, I would like to know the configuration of your setting.json and the time spent training. Thank you！

**Reply by piotrwitkowski** [reply]
For approximately 90 hours of audio on an A100 80GB, the training took around 24 hours. VRAM usage ranged between 69 and 73 GB throughout.

I used the same configuration values as in [this config file](https://github.com/SWivid/F5-TTS/blob/main/src/f5_tts/configs/F5TTS_v1_Base.yaml), except I set the tokenizer to char. Not sure which settings.json file you’re referring to — I used the training option, not fine-tuning.

Interestingly, some of my audio samples included multiple speakers within a single sentence. During inference, I could actually hear the output switch between different speakers in a single sample.

**Reply by ILG2021** [reply]
Hello, I wanna to know if you use pretrain model?

**Reply by piotrwitkowski** [reply]
No pre-trained model was used.

**Reply by ILG2021** [reply]
how many steps do you run?

**Reply by piotrwitkowski** [reply]
If I recall correctly it was around 350k steps.

**Reply by ILG2021** [reply]
So v1 base make life much easier? I am expect to try.

**Comment by ILG2021** [comment]
Hello, @piotrwitkowski  I have begin to train a 44k v1 base model again. I use bigvgan 44k vocoder, now it is go to 90k, but the sample is still has many noise. The ref sample is ok. I think the vocoder works ok. Is it because bigvgan hard to convergence? This time, I set batch size to 4000, and Gradient Accumulation Steps is 9, so the equal batch size is 36000, nearly 38400.

in the last try, I use batch size 4000 and Accumulation Steps 1, even train for 6400k, the result is bad, I don't know why.

**Reply by ILG2021** [reply]
@piotrwitkowski

**Reply by piotrwitkowski** [reply]
I would be hard to tell. I assume you train from scratch and not fine-tune? Does the model get's better over time? Do you have any inference samples? What hardware do you use?

**Reply by ILG2021** [reply]
> I would be hard to tell. I assume you train from scratch and not fine-tune? Does the model get's better over time? Do you have any inference samples? What hardware do you use?

Yes, from scratch. there is no pretrain 44k model. I am using a single 4080 GPU.

**Reply by piotrwitkowski** [reply]
May be worth running the training for much longer to see if it start converging.

**Reply by ILG2021** [reply]
The update goes to 180k, still not better. I want to make sure that you Accumulation Steps = 1, right? I have ever see the authors use 4x or 8x h100 to train from scratch.

**Reply by piotrwitkowski** [reply]
yes, accumulation step was set to 1

**Reply by SWivid** [reply]
@ILG2021 The batch size is of `frames`, so 44kHz with same number actually much fewer trained.
We use big batch and longer train to ensure zero-shot ability, but it would be acceptably shorter if infer in-set.

In general, would be better give it more patience, just wander around and go back to see if baby model grow up strong.

**Reply by ILG2021** [reply]
> @ILG2021 The batch size is of `frames`, so 44kHz with same number actually much fewer trained. We use big batch and longer train to ensure zero-shot ability, but it would be acceptably shorter if infer in-set.
> 
> In general, would be better give it more patience, just wander around and go back to see if baby model grow up strong.

Thanks, I neglect it. So for 44k, the equal batch size should double? I just train a single speaker model.

**Reply by SWivid** [reply]
> So for 44k, the equal batch size should double? I just train a single speaker model.

If you choose the 44kHz 256x version, yes. 
And it would be harder for model to learn as sequence length gap larger between text and speech.

Maybe 512x (hop length 512 if I get it right) would suit as audio compressed to similar length. Cannot say for sure as we haven't tested for this specific vocoder's quality, nor train with it.

**Reply by ILG2021** [reply]
Thanks very much for your guide. If I use 512x, should I keep Accumulation Steps to 9 or double it to 20? I use a 4080 GPU, batch size is 4000. 
Another question, If I use 256x, the batch audio sequence length is same to vocos, 512x seems easy to align with text, am I right?

The 256x config is :
target_sample_rate = 44100
n_mel_channels = 128
hop_length = 256
win_length = 1024
n_fft = 1024
mel_spec_type = "bigvgan" 

and 512x:
n_fft = 2048
hop_length = 512
win_length = 2048
n_mel_channels = 128
target_sample_rate = 44100

@SWivid

**Comment by ILG2021** [comment]
After change to bigvgan 44k 512x, things go to much better. in 5k updates, the vocal is clear. only waiting for text and speech alignment.
I use v1 base, batch size 3840, accumulate step 10, effective batch size 38400.

**Comment by ILG2021** [comment]
Hello, @piotrwitkowski what speed when you train?

**Comment by lbdave94** [comment]
Hi @piotrwitkowski do you have the chance to plot you validation loss during training? I'm training a multi language model, but I want to get an idea to what expect

**Reply by piotrwitkowski** [reply]
Hey @lbdave94 , sorry, unfortunately not.

**Reply by ILG2021** [reply]
> Hey @lbdave94 , sorry, unfortunately not.

how about speed? @piotrwitkowski

**Reply by ILG2021** [reply]
I have go to 340k, the alignment is still not very good.

**Reply by piotrwitkowski** [reply]
I don't have the numbers for my previous run, but looks like depends on the batch size: for current training (different dataset size tho) I am seeing:
38400 -> 1.1 to 1.3 updates/s
16000 ->  1.5 to 2.2 updates /s
4000 -> ~6 updates /s 
(my 350k updates for the previous model were for batch size = 38400)

**Reply by ILG2021** [reply]
24 * 3600 * 1.3 = 112 320 only 112k, seems a little strange.

**Reply by SWivid** [reply]
> 24 * 3600 * 1.3 = 112 320 only 112k, seems a little strange.

@piotrwitkowski is right, check our report.
Pretraining model takes weeks with large batchsize.

**Reply by ILG2021** [reply]
I have train bigvgan 44k 512x from scratch, equal batch size is 38400, it is about 400k now, the voice is ok now, but the alignment seems limit to 6 seconds. If the wav is 6 second in total, the align is ok. If exceed 6 second, the first part is align but the other part is a mess. What may be the cause? 

I have debug a 11s mel shape with bigvgan 44k 512x and vocos mel spec.
bigvgan 44k 512x:
torch.Size([1, 128, 983])

vocos:
torch.Size([1, 100, 1070])

@SWivid

**Reply by SWivid** [reply]
@ILG2021 what is the maximum length in your trainset?

**Reply by ILG2021** [reply]
> @ILG2021 what is the maximum length in your trainset?

15s

**Reply by ILG2021** [reply]
> > @ILG2021 what is the maximum length in your trainset?
> 
> 15s. Ok, i get it, total length is 15, if ref audio is 7.5s, it can generate 7.5s output at most.

**Comment by ILG2021** [comment]
my train has gone to 510k, the voice and the alignment are both ok now. Thanks for your help. But the audio quality is much lower than original vocoder gt, seems 70% if gt is 100%. The result is a bit lack fedelity. before  I have finetuned 24k bigvgan model, the audio quality is not that good too. So what may be the cause.
@SWivid @ZhikangNiu

**Reply by SWivid** [reply]
Better provide some examples. And quality in which aspect, word error rate, speaker similarity, utmos, or stuff?

**Reply by ILG2021** [reply]
https://drive.google.com/file/d/1P607EthXp37hxYl21vkbKox382koSeq3/view?usp=sharing
this the sample. it is not that clear and the sound is a bit muffled

**Reply by SWivid** [reply]
> https://drive.google.com/file/d/1P607EthXp37hxYl21vkbKox382koSeq3/view?usp=sharing
> this the sample. it is not that clear and the sound is a bit muffled

Have viewed the melspec of this example, but hard to judge as no ground truth wav to compare with.
From what I've observed, the high freq parts exist well, but the low freq part doesn't seem to be very clear. If you could compare the melspecs of (1; wav->tts->vocoder) with (2; wav->vocoder) of longer sample, would give more confidence. Basically I think it's scale issue (i.e. data amount and training time for 44khz)

p.s. Maybe open a separate issue with a link to here (#961) so that more potentially interested people can share some experience if they are also dealing with 44khz, and also not steer too far from the original discussion board topic.
